EIPS/eip-2364.md

```diff
@@ -0,0 +1,75 @@
+---
+eip: 2364
+title: eth/64: forkid-extended protocol handshake
+author: Péter Szilágyi <peterke@gmail.com>
+discussions-to: https://github.com/ethereum/EIPs/issues/2365
+status: Draft
+type: Standards Track
+category: Networking
+created: 2019-11-08
+requires: 2124
+---
+
+## Abstract
+
+The [`forkid` (EIP-2124)](https://eips.ethereum.org/EIPS/eip-2124) was designed to permit two Ethereum nodes to quickly and cheaply decide if they are compatible or not, not only at a genesis/networking level, but also from the perspective of the currently passed network updates (i.e. forks).
+
+EIP-2124 only defined how the `forkid` is calculated and validated, but due to the numerous possible applications, it did not itself spec out how the `forkid` should be exchanged between peers. This current EIP requests the inclusion of the `forkid` as a new field in the Ethereum wire protocol (`eth`) handshake (releasing a new version, `eth/64`).
+
+By cross validating each others' forks already during the handshake, incompatible nodes can disconnect before expensive block exchanges and validations take place (PoW check, EVM execution, state reconstruction). This furher prevents peer slots from being taken up by nodes that are incompatible, but have not yet been detected as such.
+
+## Motivation
+
+From a micro perspective, cutting off incompatible nodes from one another ensures that a node only spends its resources on tasks that are genuinely useful to it. The sooner we can decide the remote peer is useless, the less time and processing we expend in vain.
+
+From a macro perspective, keeping incompatible nodes partitioned from one another ensues that disjoint clusters retain more resources for maintaining their own chain, thus raising the quality of service for all networks globally.
+
+## Specification
+
+The specification is tiny due to the crux already being specced in EIP-2124.
+
+- Implement `forkid` generation and validation per [EIP-2124](https://eips.ethereum.org/EIPS/eip-2124).
+- Advertise a new `eth` protocol capability (version) at `eth/64`.
+  - The old `eth/63` protocol must still be kept alive side-by-side.
```

Comment:
Would be nice to have an EIP for `eth/63`... :wink:

Is there at least a spec somewhere this can link to when referring to `eth/63`?

Reply:
https://github.com/ethereum/devp2p/blob/master/caps/eth.md

Reply:
Can you put a link to it? It is hard to understand the diff without.

Reply:
Fixed in https://github.com/ethereum/EIPs/pull/2364#discussion_r344491874